tests/commands/test_build.py

```diff
@@ -2008,6 +2008,241 @@ def test_show_linters_lint_errors_forced(basic_project, emitter, config):
+    """Parts are declared with a charm part with implicit plugin.
+
+    When the "parts" section exists in chamcraft.yaml and a part named "charm"
+    is defined with implicit plugin (or explicit "charm" plugin), populate it
+    with the defaults for charm building.
+    """
+    charmcraft_yaml(
+        basic_project,
+        dedent(
+            """
+            parts:
+              charm:
+                prime:
+                  - my_extra_file.txt
+            """
+        ),
+    )
+
+    config = load(basic_project)
+    monkeypatch.chdir(basic_project)
+    builder = get_builder(config, entrypoint=None)
+
+    # create a requirements.txt file
+    pathlib.Path(basic_project, "requirements.txt").write_text("ops >= 1.2.0")
+
+    monkeypatch.setenv("CHARMCRAFT_MANAGED_MODE", "1")
+    with patch("charmcraft.parts.PartsLifecycle", autospec=True) as mock_lifecycle:
+        mock_lifecycle.side_effect = SystemExit()
+        with pytest.raises(SystemExit):
+            builder.run([0])
+    mock_lifecycle.assert_has_calls(
+        [
+            call(
+                {
+                    "charm": {
+                        "plugin": "charm",
+                        "charm-entrypoint": "src/charm.py",
+                        "charm-requirements": ["requirements.txt"],
+                        "source": str(basic_project),
+                        "prime": [
+                            "my_extra_file.txt",
+                            "src",
+                            "venv",
+                            "metadata.yaml",
+                            "dispatch",
+                            "hooks",
+                            "lib",
+                            "LICENSE",
+                            "icon.svg",
+                            "README.md",
+                        ],
+                    }
+                },
+                work_dir=pathlib.Path("/root"),
+                project_dir=basic_project,
+                ignore_local_sources=["*.charm"],
+            )
+        ]
+    )
+
+
+@pytest.mark.skipif(sys.platform == "win32", reason="Windows not [yet] supported")
+def test_parts_without_charm_part(basic_project, charmcraft_yaml, monkeypatch):
+    """Parts are declared without a charm part.
+
+    When the "parts" section exists in chamcraft.yaml and a part named "charm"
+    is not defined, process parts normally and don't invoke the charm plugin.
+    This scenario is used to use parts processing to pack a generic hooks-based
+    charm.
+    """
+    charmcraft_yaml(
+        basic_project,
+        dedent(
+            """
+            parts:
+              foo:
+                plugin: nil
+            """
+        ),
+    )
+    config = load(basic_project)
+    monkeypatch.chdir(basic_project)
+    builder = get_builder(config, entrypoint=None)
+
+    # create a requirements.txt file
+    pathlib.Path(basic_project, "requirements.txt").write_text("ops >= 1.2.0")
```

Comment:
this should not be required

Reply:
It's not required, removed from the test code.